Avo.js

```diff
@@ -270,484 +269,519 @@ _avo_invoke_meta = function _avo_invoke_meta(env, type, messages, origin) {
-    }
-    if (options.strict !== undefined) {
-      __STRICT__ = options.strict !== false;
-    }
-    if (options.reportFailureAs !== undefined) {
-      __REPORT_FAILURE_AS__ = options.reportFailureAs;
-    }
-    if (!__AVO_NOOP__ && options.inspector !== undefined) {
-      __INSPECTOR__ = options.inspector;
-    } else if (__AVO_ENV__ !== 'prod') {
-      InternalAvoLogger.warn("Avo Inspector not provided in initAvo() call");
+
+    CustomNodeJS = CustomNodeJSDestination;
+    if (__AVO_ENV__ === 'prod') {
+      CustomNodeJS.make && CustomNodeJS.make(__AVO_ENV__, "2");
+    } else if (__AVO_ENV__ === 'dev') {
+      CustomNodeJS.make && CustomNodeJS.make(__AVO_ENV__, "1");
+    } else {
+      console[__REPORT_FAILURE_AS__ || 'error']("[avo] No staging key is set for Custom Node JS. Head to destination settings in Avo to set a staging key.");
+      CustomNodeJS.make && CustomNodeJS.make(__AVO_ENV__, "");
     }
-    if (__AVO_ENV__ !== 'prod') {
-      __AVO_DEV__ = true;
+    if (__AVO_DEV__) {
+      // debug console in Avo
+      _avo_invoke_meta(__AVO_ENV__, 'init', [], 'init');
     }
+  }
+}
 
-    setSystemProperties(systemProperties);
-
-    destinationOptions = destinationOptions || {};
-
+/**
+ * Signed In: Sent when user successfully signs in or when we successfully authenticate a user.
+ *
+ * @param {object} properties - the properties associatied with this event
+ * @param {string} properties.userId_ - The value used to identify the user. Make sure it's a unique sequence of characters used to identify the user.
+ * @param {string} properties.email - The user's email
+ * @param {string} properties.authenticationMethod - Type of authentication method used by user logging in or signing up
+ *
+ * @see {@link https://www.avo.app/schemas/fwtXqAc0fCLy7b7oGW40/events/54e92613-090c-4f0b-afeb-ed720eff3422}
+ */
+export function signedIn(properties) {
+  properties = properties || {};
+  if (__AVO_DEV__) {
+    // assert properties
+    var messages = [];
+    messages = messages.concat(assertUserId_(properties.userId_));
+    messages = messages.concat(assertEmail(properties.email));
+    messages = messages.concat(assertAuthenticationMethod(properties.authenticationMethod));
+    messages = messages.concat(assertClient(sysClient));
+    messages = messages.concat(assertVersion(sysVersion));
+    messages = messages.concat(AvoAssert.assertNoAdditionalProperties("Signed In", Object.keys(properties), [
+      "userId_",
+      "email",
+      "authenticationMethod"
+    ]));
+    // debug console in Avo
     if (!__AVO_NOOP__) {
-      if (__AVO_ENV__ === 'prod') {
-      }
-      if (__AVO_ENV__ === 'dev') {
-      }
+      _avo_invoke(__AVO_ENV__, "54e92613-090c-4f0b-afeb-ed720eff3422", "2fec10106ad560f43f307ffdc820e3bd7a474585f9c62ee39961730c35694cb1", messages.map(function(m) { return Object.assign({}, {tag: m.tag, propertyId: m.propertyId, additionalProperties: m.additionalProperties, actualType: m.actualType}); }), 'event');
```

Comment:
What magic strings are these and why are they changing?

Reply:
The first one is the `eventId` in Avo, the second one is the `eventHash` which we use in the implementation status to confirm if we've seen this exact version of the event sent or not.

Reply:
Ahh, I see!